bigtable/internal/testproxy/proxy_test.go

```diff
@@ -0,0 +1,201 @@
+	if err != nil {
+		return fmt.Errorf("testproxy setup: can't dial inmem Bigtable server: %v", err)
+	}
+	defer conn.Close()
+
+	adminClient, err := bigtable.NewAdminClient(ctx, "client", "instance", option.WithGRPCConn(conn), option.WithGRPCDialOption(grpc.WithBlock()))
+	if err != nil {
+		return fmt.Errorf("testproxy setup: can't create AdminClient: %v", err)
+	}
+	defer adminClient.Close()
+
+	if err := adminClient.CreateTable(ctx, tableName); err != nil {
+		return fmt.Errorf("testproxy setup: can't create table: %v", err)
+	}
+
+	// Create column families
+	count := 3
+	for i := 0; i < count; i++ {
+		cfName := fmt.Sprintf("%s%d", columnFamily, i)
+		if err := adminClient.CreateColumnFamily(ctx, tableName, cfName); err != nil {
+			return fmt.Errorf("testproxy setup: can't create column family: %s", cfName)
+		}
+	}
+
+	// Create rows
+	dataClient, err := bigtable.NewClient(ctx, "client", "instance", option.WithGRPCConn(conn), option.WithGRPCDialOption(grpc.WithBlock()))
+	if err != nil {
+		return fmt.Errorf("testproxy setup: can't create Bigtable client: %v", err)
+	}
+	defer dataClient.Close()
+
+	t := dataClient.Open(tableName)
+
+	for fc := 0; fc < count; fc++ {
+		for cc := count; cc > 0; cc-- {
+			for tc := 0; tc < count; tc++ {
+				rmw := bigtable.NewReadModifyWrite()
+				rmw.AppendValue(fmt.Sprintf("%s%d", columnFamily, fc), fmt.Sprintf("coll%d", cc), []byte("test data"))
+
+				_, err = t.ApplyReadModifyWrite(ctx, rowKey, rmw)
+				if err != nil {
+					return fmt.Errorf("testproxy setup: failure populating row: %v", err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+/*
+TestMain has three threads that it needs to start:
+1) The mocked Bigtable service (server)
+2) The NewCloudBigtableV2TestProxyClient client that sends requests to the testproxy server.
+3) The testproxy server under test
+
+	The communication sequence looks kind of like:
+
+	TestProxyClient <=> test proxy server (what we're testing) <=> mocked BT server
+*/
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+
+	// 1) Start the mocked Bigtable service
+	// This requires creating a "table" in memory
+	bts, err := bttest.NewServer(bigtableAddress)
+	if err != nil {
+		log.Fatalf("testproxy setup: can't create inmem Bigtable server")
+	}
+	err = populateTable(bts)
+	if err != nil {
+		log.Fatalf("testproxy setup: can't populate mock table")
+	}
+
+	// 3) Start the test proxy server
+	lis = bufconn.Listen(buffer)
+	s := newProxyServer(lis)
+	defer s.Stop()
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			log.Fatalf("failed to serve: %v", err)
+		}
+	}()
+
+	// 2) Create the test proxy client
+	conn2, err := grpc.DialContext(ctx, lis.Addr().String(),
+		grpc.WithContextDialer(bufDialer),
+		grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		log.Fatalf("testproxy setup: failed to dial testproxy: %v", err)
+	}
+	defer conn2.Close()
+	c := pb.NewCloudBigtableV2TestProxyClient(conn2)
```

Comment:
``` golang
c := pb.NewCloudBigtableV2TestProxyClient(conn2) 
_, err = c.CreateClient(ctx, req)
```

Reply:
I simplified some of this code, but I still require the assignment (since `client` is declared in the outer scope and I can't assign directly to `client`).